pyanaconda/modules/security/certificates/installation.py

```diff
@@ -0,0 +1,84 @@
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os
+
+from pyanaconda.anaconda_loggers import get_module_logger
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.errors.installation import SecurityInstallationError
+from pyanaconda.core.path import make_directories, join_paths
+from pyanaconda.core.constants import PAYLOAD_TYPE_DNF, INSTALLATION_PHASE_PREINSTALL
+
+log = get_module_logger(__name__)
+
+
+class ImportCertificatesTask(Task):
+    """Task for importing certificates into a system."""
+
+    def __init__(self, sysroot, certificates, payload_type=None, phase=None):
+        """Create a new certificates import task.
+
+        :param str sysroot: a path to the root of the target system
+        :param certificates: list of certificate data holders
+        :param payload_type: a type of the payload
+        :param phase: installation phase - INSTALLATION_PHASE_PREINSTALL or None for any other
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._certificates = certificates
+        self._payload_type = payload_type
+        self._phase = phase
+
+    @property
+    def name(self):
+        return "Import CA certificates"
+
+    def _dump_certificate(self, cert, root):
+        """Dump the certificate into specified file and directory."""
+        if not cert.dir:
+            raise SecurityInstallationError(
+                "Certificate destination is missing for {}".format(cert.filename)
+            )
+
+        dst_dir = join_paths(root, cert.dir)
+        if not os.path.exists(dst_dir):
+            log.debug("Path %s for certificate does not exist, creating.", dst_dir)
+            make_directories(dst_dir)
+
+        dst = join_paths(dst_dir, cert.filename)
+
+        if os.path.exists(dst):
+            log.warning("Certificate file %s already exists, replacing.", dst)
+
+        with open(dst, 'w') as f:
+            f.write(cert.cert)
+            f.write('\n')
+
+    def run(self):
+        """Import CA certificates.
+
+        Dump the certificates into specified files and directories.
+        """
+        if self._phase == INSTALLATION_PHASE_PREINSTALL:
+            if self._payload_type != PAYLOAD_TYPE_DNF:
+                log.debug("Not importing certificates in pre install for %s payload.",
+                          self._payload_type)
+                return
+
+        for cert in self._certificates:
+            log.debug("Importing certificate %s", cert)
```

Comment:
Does this log also the content of the certificate itself? I wonder if it is not enough to log just the certificate directory/filename for readability of the log file.

Reply:
Good point, I'll update the message.

Reply:
Done